Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,8 +190,49 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+
+            // check for a invalid index             
+            if (index < 0)
+            {
+                return array;
+            }
+
+            // check for a invalid array 
+            if (array == null || array.Length == 0)
+            {
+                if (index == 0)
+                {
+                    return new[] { number };
+                }
+                return array;
+            }
+
+            //  a new array that holds the new data with the new item 
+            var newArray = new int[array.Length + 1];
+            
+            // The actual copy and insert data into the new array 
+            for (int i = 0; i <= array.Length - 1; i++)
+            {
+                if (i < index)
```

Comment:
This is not so obvious

Reply:
This function is commented better than the other one :)

Reply:
Must I change this. so make more comments in this function ?

Reply:
I would say make less comments, but for better key points. You don't have to change this.